solution

BastovOleksandr opened the pull request.

BastovOleksandr committed:
solution

BastovOleksandr committed:
double inner quotes fix

BastovOleksandr committed:
config.json not found error fix

BastovOleksandr committed:
config.json not found error fix

oliasymak24 commented on app/customer.py:
```diff
@@ -0,0 +1,66 @@
+        name: str,
+        product_cart: dict,
+        location: list,
+        money: float,
```
For money usually better to use Decimal module

oliasymak24 commented on app/customer.py:
```diff
@@ -0,0 +1,66 @@
+        self.product_cart = product_cart
+        self.location = location
+        self.money = money
+        self.car = Car(**car)
```
cars will be better

oliasymak24 commented on app/customer.py:
```diff
@@ -0,0 +1,66 @@
+
+    def count_prod_cost(self, prices: dict) -> float:
+        return sum(
+            a * b for a, b in zip(self.product_cart.values(), prices.values())
```
What is `a` and `b`, choose better variable names

BastovOleksandr committed:
now Decimal's everywhere if it's money context

"car_characteristics" name for Customer argument instead of "car"

obscure variables received understandable names

Oleksl888 commented on app/main.py:
```diff
@@ -1,3 +1,36 @@
+    customers, shops = [], []
+    fuel_price = Decimal(str(trip_data["FUEL_PRICE"]))
+
+    for cust_characteristics in trip_data["customers"]:
```
Three for loops over the same data is a bit too much. You can reduce it to two loops for efficeincy. I would recommend to create Shops first, and then iterate over same data set to create customers, cars and do other things

Oleksl888 commented on app/main.py:
```diff
@@ -1,3 +1,36 @@
+    fuel_price = Decimal(str(trip_data["FUEL_PRICE"]))
+
+    for cust_characteristics in trip_data["customers"]:
+        cust_characteristics["car_characteristics"] = (
```
cust_charactersitics is a very weird name for a variable. customer_data or similar would be better

Oleksl888 commented on app/customer.py:
```diff
@@ -0,0 +1,74 @@
+            car: Car,
+            fuel_price: Decimal
+    ) -> Decimal:
+        distance = (((self.location[0] - to_location[0]) ** 2
```
I think you could implment a class Location that will have coords as parameters on init and will have calculate_distance method that will do this block of code. This will structure your code better imo

Oleksl888 commented on app/customer.py:
```diff
@@ -0,0 +1,74 @@
+        cheapest_shop = min(all_shops_total, key=all_shops_total.get)
+        return cheapest_shop, all_shops_total[cheapest_shop]
+
+    def count_fuel_cost(
```
this would be rather a Car method - not a customer method. it doesn't depend on a customer in any way so it would be logical to move it out. Your customer already has a lot of methods and having one class that does it all is not always a good idea

BastovOleksandr committed:
fixes
